daemon/list_test.go

```diff
@@ -140,3 +181,51 @@ func TestNameFilter(t *testing.T) {
 	assert.Assert(t, is.Len(containerListWithPrefix, 1))
 	assert.Assert(t, containerListContainsName(containerListWithPrefix, three.Name))
 }
+
+func TestContainerList_LimitFilter(t *testing.T) {
+	db, err := container.NewViewDB()
+	assert.NilError(t, err)
+	d := &Daemon{
+		containersReplica: db,
+	}
+
+	// start containers
+	num := 32
+	containers := make([]*container.Container, num)
+	for i := range num {
+		name := fmt.Sprintf("cont-%d", i)
+		containers[i] = setupContainerWithName(t, name, d)
+	}
```

Comment:
Just an idea; would it be good to check the actual list in the containersReplica db (as that's the source of truth used)?

```go
	const num = 32
	for i := range num {
		setupContainerWithName(t, fmt.Sprintf("cont-%d", i), d)
	}
	containers, err := db.Snapshot().All()
	assert.NilError(t, err)
	assert.Assert(t, is.Len(containers, num))
```

Reply:
Sure, good idea.